core/config/capabilities_config.go

```diff
@@ -11,8 +11,24 @@ type CapabilitiesExternalRegistry interface {
 	RelayID() types.RelayID
 }
 
+type GatewayConnectorConfig interface {
+	ChainIDForNodeKey() string
+	NodeAddress() string
+	DonID() string
+	Gateways() []ConnectorGatewayConfig
+	WsHandshakeTimeoutMillis() uint32
+	AuthMinChallengeLen() int
+	AuthTimestampToleranceSec() uint32
+}
+
+type ConnectorGatewayConfig interface {
+	ID() string
+	URL() string
+}
+
 type Capabilities interface {
 	Peering() P2P
 	Dispatcher() Dispatcher
 	ExternalRegistry() CapabilitiesExternalRegistry
+	GatewayConnectorConfig() GatewayConnectorConfig
```

Comment:
In most places, the `Config` suffix is redundant:
```suggestion
type GatewayConnector interface {
	ChainIDForNodeKey() string
	NodeAddress() string
	DonID() string
	Gateways() []ConnectorGateway
	WsHandshakeTimeoutMillis() uint32
	AuthMinChallengeLen() int
	AuthTimestampToleranceSec() uint32
}

type ConnectorGateway interface {
	ID() string
	URL() string
}

type Capabilities interface {
	Peering() P2P
	Dispatcher() Dispatcher
	ExternalRegistry() CapabilitiesExternalRegistry
	GatewayConnector() GatewayConnector
```